src/rust_api.rs

```diff
@@ -15,6 +15,7 @@ pub trait BaseSequence: std::marker::Sized {
 
     fn as_slice(&self) -> &[Self::Item];
     fn as_mut_slice(&mut self) -> &mut [Self::Item];
+    fn push(&mut self, _item: Self::Item) {}
```

Comment:
I don't like this having a default impl that does nothing. It seems error-prone.

I'd prefer that `push` on `ProteinSequence` and `DnaSequence` are just `pub fn`s in impl blocks for those types, like how `DnaSequence` already has `translate`. Sort of like how `Vec` and `BinaryHeap` both have `push`, but it's not part of a trait, just an associated method with similar semantics & name. We might want to implement `BaseSequence` for immutable view types at some point, and having a mutable function in the trait will be a blocker for that.

Reply:
Ah, this is a typo, I didn't intend for a default impl.

I'm fine w/ not having `push` as a part of the trait

Reply:
ping @vgel